src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -628,6 +628,30 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
             return (results, page.ContinuationToken);
         }
 
+        internal async Task<IReadOnlyList<T>> ExecutePagedQueryAsync<T>(QueryDefinition sqlQuerySpec, QueryRequestOptions feedOptions, CancellationToken cancellationToken = default)
```

Comment:
I'd like to talk about this in our sync. If this fixes the order by problem I'd like to take it into bulk delete.